--- sorcer/CMakeLists.txt-old	2020-11-20 19:21:25.915381197 -0500
+++ sorcer/CMakeLists.txt	2020-11-20 19:22:44.298305200 -0500
@@ -24,9 +24,15 @@
 
 SET(CMAKE_SHARED_LINKER_FLAGS "-fPIC -shared -Wl,-z,nodelete -Wl,--no-undefined -lXft -lX11 -lpthread -lfontconfig -lfreetype -ldl")
 
+MESSAGE("PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
+
+IF(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+  SET(MACHINE_OPT_FLAGS "-msse2 -mfpmath=sse")
+ENDIF()
+
 IF(RELEASE_BUILD)
-  SET(CMAKE_CXX_FLAGS "-g -Wall -Wno-unused-variable -msse2 -mfpmath=sse -ffast-math")
-  SET(CMAKE_C_FLAGS "-g -Wall -W -Wno-unused-variable -msse2 -mfpmath=sse -ffast-math")
+  SET(CMAKE_CXX_FLAGS "-g -Wall -Wno-unused-variable ${MACHINE_OPT_FLAGS} -ffast-math")
+  SET(CMAKE_C_FLAGS "-g -Wall -W -Wno-unused-variable ${MACHINE_OPT_FLAGS} -ffast-math")
 ENDIF()
 
 ## The following commands will re-generate the .cpp files from the FAUST .dsp,
